backend/modules/lobby/controllers/generation.ts

```diff
@@ -22,9 +22,9 @@ export const generateCodeAndLobby = async (
 ) => {
   try {
     const userIdFromToken = req.user?.id;
-    if (!userIdFromToken) {
-      throw new Error("User ID is missing");
-    }
+    // if (!userIdFromToken) {
```

Comment:
This check is already performed by middleware, it's not doing anything additionally. Also go through all the controllers and remove checks that are redundant